packages/middleware-retry/src/defaultStrategy.ts

```diff
@@ -35,12 +35,12 @@ export interface DelayDecider {
 
 export class ExponentialBackOffStrategy implements RetryStrategy {
   constructor(
-    public readonly maxRetries: number,
+    public readonly maxAttempts: number,
     private retryDecider: RetryDecider = defaultRetryDecider,
     private delayDecider: DelayDecider = defaultDelayDecider
   ) {}
   private shouldRetry(error: SdkError, retryAttempted: number) {
-    return retryAttempted < this.maxRetries && this.retryDecider(error);
+    return retryAttempted < this.maxAttempts && this.retryDecider(error);
```

Comment:
See my comment above, `retryAttempted` should be `attempts` and should start from `1`, or `attempts` should self-increase before calling `shouldRetry()`(line 60).

Reply:
Fixed in [commit](https://github.com/aws/aws-sdk-js-v3/pull/1244/commits/7d63dea111ba0e0f880ad309406f3165a613b43d)